keps/sig-auth/2907-secrets-store-csi-driver/README.md

```diff
@@ -0,0 +1,368 @@
+
+### Graduation Criteria
+
+#### GA
+
+- Month+ soak of minor release
+- Completion of milestone requirements
+- Agreement of stability documented on community call from 3+ provider maintainers
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+This will be deployed to clusters as a standalone separate component.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+- non-zero `total_node_publish` metrics indicate the CSI driver is used by the workloads.
+- `total_sync_k8s_secret` metrics indicate the optional Sync as Kubernetes secret feature is used by the workloads.
+- `total_rotation_reconcile` metrics indicate the optional rotation reconciliation feature is used by the workloads.
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
```

Comment:
failure in pod creation?

Reply:
I dont think we can get a time-series metric for that from the `kubelet` or the `kube-apiserver`, but some kubelet measurement of failed volume mounts would be useful as there is an error mode where the request does not make it to the driver process because of issues registering the driver with the kubelet.

We could maybe use `kubectl get events --field-selector="reason=FailedMount" --all-namespaces` but that can be hard to monitor and there does not appear to be an `event` for _successful_ `Mount` operations.